src/browser/connection/gateway.ts

```diff
@@ -245,6 +247,30 @@ export default class BrowserConnectionGateway {
             respond500(res, 'There are no available _connections to establish.');
     }
 
+    private static _parseSelector (req: IncomingMessage, res: ServerResponse, connection: BrowserConnection): void {
+        if (BrowserConnectionGateway._ensureConnectionReady(res, connection)) {
+            BrowserConnectionGateway._fetchRequestData(req, data => {
+                try {
+                    const options = {
```

Comment:
Pack this logic to the separate function: `_getParsedSelector()`.

Reply:
https://github.com/DevExpress/testcafe/pull/7372/commits/6f1b0ad5a1266f8711260da6b24177ecd7637dc5